infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/SenderBufferManager.java

```diff
@@ -169,36 +170,48 @@ public LogReplicationEntry processAcks() throws InterruptedException, ExecutionE
         return ack;
     }
 
-    public CompletableFuture<LogReplicationEntry> sendWithBuffering(LogReplicationEntry message) {
-        message.getMetadata().setSnapshotSyncSeqNum(snapshotSyncSequenceNumber++);
-        pendingMessages.append(message);
-        CompletableFuture<LogReplicationEntry> cf = dataSender.send(message);
+    public CompletableFuture<LogReplicationEntryMsg> sendWithBuffering(LogReplicationEntryMsg message) {
+        LogReplicationEntryMetadata metadata = LogReplicationEntryMetadata.newBuilder()
+                .mergeFrom(message.getMetadata())
+                .setSnapshotSyncSeqNum(snapshotSyncSequenceNumber++).build();
+
+        LogReplicationEntryMsg newMsg = LogReplicationEntryMsg.newBuilder()
+                .mergeFrom(message)
+                .setMetadata(metadata)
+                .build();
+        pendingMessages.append(newMsg);
```

Comment:
Same here

Reply:
Extracted this into `overrideMetadata()`.